src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/ArgHelper.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+#if !NET8_0_OR_GREATER
+using System.Collections.Generic;
+#endif
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Razor;
+
+internal static class ArgHelper
+{
+    public static void ThrowIfNull([NotNull] object? argument, [CallerArgumentExpression(nameof(argument))] string? paramName = null)
+    {
+#if NET8_0_OR_GREATER
+        ArgumentNullException.ThrowIfNull(argument, paramName);
+#else
+        if (argument is null)
+        {
+            ThrowException(paramName);
+        }
+
+        [DoesNotReturn]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void ThrowException(string? paramName)
```

Comment:
Nit: Why not just call the `ThrowArgumentNullException` method?

Reply:
Thanks! I missed this.